.github/workflows/lfric_test.yml

```diff
@@ -74,40 +74,93 @@ jobs:
         export PSYCLONE_LFRIC_DIR=${GITHUB_WORKSPACE}/examples/lfric/scripts
-        export MINIAPP_DIR=${HOME}/LFRic/miniapps/gungho_model
+        export MINIAPP_DIR=${HOME}/LFRic_45376/miniapps/gungho_model
+        export PSYCLONE_CONFIG=${HOME}/lfric_psyclone_config.cfg
         cd ${MINIAPP_DIR}
         # Compile
-        module load lfric_env
-        export PSYCLONE_CONFIG=${HOME}/lfric_psyclone_config.cfg
         make clean
         make -j 6 build
         # Run
         cd example
-        cp ${HOME}/lfric_gunho_configuration.nml configuration.nml
-        mpirun -n 6 ../bin/gungho_model configuration.nml
-        tail PET0.gungho_model.Log
+        cp ${HOME}/lfric_gunho_configuration_4its.nml configuration.nml
+        mpirun -n 1 ../bin/gungho_model configuration.nml |& tee output.txt
+        python ${PSYCLONE_LFRIC_DIR}/compare_ouput.py ${HOME}/lfric_gunho_configuration_4its_output.txt output.txt
         cat timer.txt
+        export VAR_TIME=$(grep "gungho_model" timer.txt | cut -d'|' -f5)
+        export VAR_HALOS=$(grep "gungho_model" halo_calls_counter.txt | cut -d'|' -f5)
+        echo $GITHUB_REF_NAME $GITHUB_SHA $VAR_TIME $VAR_HALOS >> ${HOME}/lfric_passthrough_performance_history
+        # Upload the results into a Github Gist
+        newresults=$(printf "%q" "$(cat ${HOME}/lfric_passthrough_performance_history)")
+        {
+          printf '{"description":"New LFRic results","files":{"lfric_passthrough.txt":{"content":"'
+          echo -n ${newresults:2:-1}
+          printf '"}}}'
+        }| curl -L -X PATCH \
+             -H "Accept: application/vnd.github+json" \
+             -H "Authorization: Bearer ${{ secrets.GIST_TOKEN }}" \
+             -H "X-GitHub-Api-Version: 2022-11-28" \
+             https://api.github.com/gists/a4049a0fc0a0a11651a5ce6a04d76160 \
+             --data-binary @-
 
     # PSyclone, compile and run MetOffice LFRic with all optimisations and 6 OpenMP threads
     - name: LFRic with all transformations
       run: |
-        . .runner_venv/bin/activate
+        # Set up environment
+        source /apps/spack/spack-repo/share/spack/setup-env.sh
+        spack env activate lfric-gnu
+        export FC="gfortran"
+        export CC="gcc"
+        export CXX="g++"
+        export FPP="cpp -traditional-cpp"
+        export LDMPI="mpif90"
+        export FFLAGS="-I$(spack location -e lfric-gnu)/.spack-env/view/include"
+        export LDFLAGS="-L$(spack location -e lfric-gnu)/.spack-env/view/lib"
+        export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$(spack location -e lfric-gnu)/.spack-env/view/lib"
+        export LD_LIBRARY_PATH="$LD_LIBRARY_PATH::$(spack location -e lfric-gnu)/.spack-env/view/lib64"
+        source .runner_venv/bin/activate
         export PSYCLONE_LFRIC_DIR=${GITHUB_WORKSPACE}/examples/lfric/scripts
-        export MINIAPP_DIR=${HOME}/LFRic/miniapps/gungho_model
+        export MINIAPP_DIR=${HOME}/LFRic_45376/miniapps/gungho_model
```

Comment:
We could definitely parameterise the location of the LFRic source as an environment variable.

Reply:
Again, home is not accessible from env, but I think it make sense to parametrise the revision number for future updates.

Reply:
Great. I think that is what I was after.